src/app/rundown/components/segment/segment.component.ts

```diff
@@ -91,6 +94,11 @@ export class SegmentComponent implements OnChanges, OnDestroy {
     this.timeReference = timeSpendUntilActivePart + timeSpendInActivePart
   }
 
+  public setFirstPartAsNext(): void {
+    const firstPart: Part = this.segment.parts[0]
```

Comment:
Technically a Segment can have 0 parts, thus this _could_ return undefined. 
Don't know if this is something that is handled elsewhere, or even if it should be handled on the frontend.

Reply:
For now i've added handling for the first part not being set as next if it contains no pieces. Whether the handling should extend further into scenarios where we e.g. have a first part as invalid and then should skip to the one after, is not considered. But i don't know if that is even relevant at this point in time.

Reply:
Perhaps it should search for the first valid part, if any, instead.

Reply:
I have changed it to find a valid part instead!